lib/WarnCaseSensitiveModulesPlugin.js

```diff
@@ -2,27 +2,30 @@
 	MIT License http://www.opensource.org/licenses/mit-license.php
 	Author Tobias Koppers @sokra
 */
-var CaseSensitiveModulesWarning = require("./CaseSensitiveModulesWarning");
+"use strict";
 
-function WarnCaseSensitiveModulesPlugin() {}
-module.exports = WarnCaseSensitiveModulesPlugin;
+const CaseSensitiveModulesWarning = require("./CaseSensitiveModulesWarning");
 
-WarnCaseSensitiveModulesPlugin.prototype.apply = function(compiler) {
-	compiler.plugin("compilation", function(compilation) {
-		compilation.plugin("seal", function() {
-			var moduleWithoutCase = {};
-			this.modules.forEach(function(module) {
-				var ident = module.identifier().toLowerCase();
-				if(moduleWithoutCase["$" + ident]) {
-					moduleWithoutCase["$" + ident].push(module);
-				} else {
-					moduleWithoutCase["$" + ident] = [module];
-				}
-			}, this);
-			Object.keys(moduleWithoutCase).forEach(function(key) {
-				if(moduleWithoutCase[key].length > 1)
-					this.warnings.push(new CaseSensitiveModulesWarning(moduleWithoutCase[key]));
-			}, this);
+class WarnCaseSensitiveModulesPlugin {
+	apply(compiler) {
+		compiler.plugin("compilation", compilation => {
+			compilation.plugin("seal", () => {
+				const moduleWithoutCase = {};
```

Comment:
@sokra @Kovensky I tried it before and the related code with the changes will be something like. 

```javascript
...
compiler.plugin("compilation", compilation => {
	compilation.plugin("seal", () => {
		const moduleWithoutCase = new Map();
		compilation.modules.forEach(module => {
			const identifier = module.identifier().toLowerCase();
			const moduleInstances = moduleWithoutCase.get(identifier);
			if(moduleInstances) {
				moduleInstances.push(module);
				moduleWithoutCase.set(identifier, moduleInstances);
			} else {
				moduleWithoutCase.set(identifier, [module]);
			}
		});
		Object.keys(moduleWithoutCase).forEach(key => {
			const moduleInstances = moduleWithoutCase.get(key);
			if(moduleInstances.length > 1)
				compilation.warnings.push(new CaseSensitiveModulesWarning(moduleInstances));
		});
	});
});
...
```

By the way, it's not working with those changes. Do you guys know why? Can you guys help me with that because I tried to debug our tests, but it's been hard. Any ideas?